src/Writer/Extension/PodcastIndex/Renderer/Entry.php

```diff
@@ -94,22 +117,261 @@ protected function setChapters(DOMDocument $dom, DOMElement $root): void
         if (! $soundbites) {
             return;
         }
         foreach ($soundbites as $soundbite) {
-            /** @psalm-var array<string, string> $soundbite */
-            $el = $dom->createElement('podcast:soundbite');
-            if (array_key_exists('title', $soundbite)) {
-                $text = $dom->createTextNode((string) $soundbite['title']);
-                $el->appendChild($text);
-            }
-            $el->setAttribute('startTime', $soundbite['startTime']);
-            $el->setAttribute('duration', $soundbite['duration']);
+            $el = ElementGenerator::createPodcastIndexElement($dom, $soundbite, 'soundbite');
             $root->appendChild($el);
             $this->called = true;
         }
     }
+
+    /**
+     * Set entry location
+     */
+    private function setLocation(DOMDocument $dom, DOMElement $root): void
+    {
+        /** @psalm-var EntryWriter $container */
+        $container = $this->getDataContainer();
+
+        /** @psalm-var null|LocationArray $location */
+        $location = $container->getPodcastIndexLocation();
+        if ($location === null) {
+            return;
+        }
+        $el = ElementGenerator::createPodcastIndexElement($dom, $location, 'location', 'description');
+        $root->appendChild($el);
+        $this->called = true;
+    }
+
+    /**
+     * Set feed license
+     */
+    private function setLicense(DOMDocument $dom, DOMElement $root): void
+    {
+        /** @psalm-var EntryWriter $container */
+        $container = $this->getDataContainer();
+
+        /** @psalm-var null|LicenseArray $license */
+        $license = $container->getPodcastIndexLicense();
+        if ($license === null) {
+            return;
+        }
+        $el = ElementGenerator::createPodcastIndexElement($dom, $license, 'license', 'identifier');
+
+        $root->appendChild($el);
+        $this->called = true;
+    }
+
+    /**
+     * Set feed people
+     */
+    private function setPeople(DOMDocument $dom, DOMElement $root): void
+    {
+        /** @psalm-var EntryWriter $container */
+        $container = $this->getDataContainer();
+
+        /** @psalm-var null|list<PersonArray> $people */
+        $people = $container->getPodcastIndexPeople();
+        if ($people === null || $people === []) {
+            return;
+        }
+        foreach ($people as $person) {
+            $el = ElementGenerator::createPodcastIndexElement($dom, $person, 'person', 'name');
+            $root->appendChild($el);
+        }
+        $this->called = true;
+    }
+
+    /**
+     * Set entry txts
+     */
+    private function setTxts(DOMDocument $dom, DOMElement $root): void
+    {
+        /** @psalm-var EntryWriter $container */
+        $container = $this->getDataContainer();
+
+        /** @psalm-var list<TxtArray>|null $txts */
+        $txts = $container->getPodcastIndexTxts();
+        if ($txts === null || $txts === []) {
+            return;
+        }
```

Comment:
Case in point, if `getPodcastIndexTxts` only returns an array, you don't need to check for null here